tests/test_packages/test_skills/test_carpark_detection/test_database.py

```diff
@@ -0,0 +1,383 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the tests of the RegistrationDB class of the carpark detection skill."""
+import os
+import sqlite3
+from pathlib import Path
+from unittest.mock import Mock, patch
+
+from aea.test_tools.test_skill import BaseSkillTestCase
+
+from packages.fetchai.skills.carpark_detection.database import DetectionDatabase
+
+from tests.conftest import ROOT_DIR
+
+
+class TestDetectionDatabase(BaseSkillTestCase):
+    """Test DetectionDatabase of carpark detection."""
+
+    path_to_skill = Path(ROOT_DIR, "packages", "fetchai", "skills", "carpark_detection")
+
+    @classmethod
+    def setup(cls):
+        """Setup the test class."""
+        super().setup()
+        cls.temp_dir = os.path.join(os.path.dirname(__file__), "temp_files_placeholder")
+        cls.create_if_not_present = True
+        cls.logger = None
+        cls.db = DetectionDatabase(
+            temp_dir=cls.temp_dir,
+            create_if_not_present=cls.create_if_not_present,
+            logger=cls.logger,
+        )
+
+    def test__init__i(self):
+        """Test the __init__ method of the DetectionDatabase class."""
+        # operation
+        with patch.object(self.db, "initialise_backend") as mock_initialise_back:
+            self.db.__init__(self.temp_dir, self.create_if_not_present, self.logger)
+
+        mock_initialise_back.assert_called_once()
+
+    def test_is_db_exits_i(self):
```

Comment:
There seems to be a typo in the function name in the car park skill. I guess that  `is_db_exits` was meant to be `is_db_exists`, which is kind of strange grammar (probably `does_db_exist` or just `db_exists` would be better). Anyway, this seems worth the minor name change, since it is a bit confusing.